dashboard/utils.py

```diff
@@ -292,35 +283,139 @@ def attendance_list(request, event):
-        semester.save()
-
-    if date.weekday() >= instance['meeting_day']:
-        date_offset = 7 + instance['meeting_day'] - date.weekday()
-    elif date.weekday() < instance['meeting_day']:
-        date_offset = instance['meeting_day'] - date.weekday()
+def semester_end_date(season, year, treat_summer_fall_as_same=True):
+    """Calculate the last day of the semester given its season and year.
+
+    The last day of the semester is the 31st of its ending month
+    (May for Spring, July for Summer, December for Fall)
+
+    :param str season:
+        '0', '1', or '2' to represent Spring, Summer, or Fall respectively
+    :param int year:
+        The year of the semester to find the end date of
+    :param bool treat_summer_fall_as_same:
+        If true and summer is the season, it will be treated as though fall
+        were given rather than summer
+    :returns:
+        the end date of the semester
+    :rtype: datetime
+
+    """
+    # if spring semester, use may as the end month
+    if season == '0':
+        end_month = 5
+    elif season == '1' and not treat_summer_fall_as_same:
+        end_month = 7
+    # otherwise, we are in the summer or fall semester, so use december as end month
+    else:
+        end_month = 12
+
+    # last day of both May, July, and December is the 31st.
+    last_day_of_month = 31
+
+    return datetime.datetime(year, end_month, last_day_of_month)
+
+
+def semester_start_date(season, year):
+    """Calculate the start date of a semester given a season and year.
+
+    The Start date of a semester is the first day of the month in which is starts
+    (January for Spring, June for Summer, August for Fall)
+
+    :param str season:
+        '0', '1', or '2' to represent Spring, Summer, or Fall respectively
+    :param int year:
+        The year of the semester to find the start date of
+    :returns:
+        the start date of the semester
+    :rtype: datetime
+
+    """
+    # Find the start month for the semester season
+    start_month = {
+        '0': 1,
+        '1': 6,
+        '2': 8
+    }.get(season)
+
+    start_day_of_month = 1
+
+    return datetime.datetime(year, start_month, start_day_of_month)
+
+
+def create_recurring_events(begin_date, day, interval, event_creator):
+    """Create recurring events for the semester begin_date falls in.
+
+    :param datetime begin_date:
+        The earliest date at which an event could be scheduled.  Events are
+        schedule only during the same semester as begin_date.
+    :param int day:
+        The integer day of the week on which the recurring events happen
+    :param int interval:
+        The number of day in between each event
+    :param function event_creator:
+        A function that takes (datetime, Semester) and creates an event.  The
+        function does not need to return a value (any value returned is ignored).
+
+    """
+    date = begin_date
+    semester = get_semester(date)
+
+    # offset to the next week
```

Comment:
```suggestion
    # offset to the start of the next week
```

Just for clarity's sake: The purpose of this is to change the date to the start of the week so that it can start creating recurring events on the first day that they can occur. As such, the first if statement checks if the given day has already passed this week (for instance if this week Thursday has already passed). If so, it chooses the starting week as next week. Otherwise, if Thursday hasn't passed yet, (the elif) the starting week is this week. Also, you might be able to change the elif to an else.